tasks/test.py

```diff
@@ -124,37 +124,51 @@ def install_tools(ctx):
                     ctx.run(f"go install {tool}")
 
 
-# TODO(AP-1879): The following four functions all do something similar: they run a given command on a list of modules
-# This could be refactored in a core function that does the loop on modules and returns failures, and
-# wrapper functions that craft the command to run and process the results and errors.
-
-
-def lint_flavor(
-    ctx, modules: List[GoModule], flavor: AgentFlavor, build_tags: List[str], arch: str, rtloader_root: bool
+def test_core(
+    modules: List[GoModule],
+    flavor: AgentFlavor,
+    module_class: GoModule,
+    flavor_print_string: str,
+    command,
+    skip_module_class=False,
 ):
     """
-    Runs linters for given flavor, build tags, and modules.
+    Run the command function on each module of the modules list.
     """
-    modules_lint_results = []
-    print(f"--- Flavor {flavor.name}: golangci_lint")
+    modules_results = []
+    print(f"--- Flavor {flavor.name}: {flavor_print_string}")
```

Comment:
```suggestion
    print(f"--- Flavor {flavor.name}: {operation_name}")
```

Reply:
Fixed in [b4540b7](https://github.com/DataDog/datadog-agent/pull/15957/commits/b4540b7e10da28b6052cd07029be7a14f9bb7213)